src/wallet/wallet.cpp

```diff
@@ -2060,6 +2082,59 @@ void CWallet::AvailableCoins(std::vector<COutput>& vCoins, bool fOnlySafe, const
     }
 }
 
+std::map<CTxDestination, std::vector<COutput>> CWallet::ListCoins() const
+{
+    std::map<CTxDestination, std::vector<COutput>> result;
```

Comment:
nit: might be nice to AssertLockHeld here (because you cant call this without cs_wallet due to the COutput returns).